src/xo-app/dashboard/overview/index.js

```diff
@@ -18,10 +25,131 @@ import {
   formatSize
 } from 'utils'
 import {
+  getHostMissingPatches,
+  installAllHostPatches,
   isSrWritable,
   subscribeUsers
 } from 'xo'
 
+import styles from './index.css'
+
+// ===================================================================
+
+@connectStore(() => {
+  const getPools = createGetObjectsOfType('pool')
+  const getHosts = createGetObjectsOfType('host').sort()
+
+  return {
+    pools: getPools,
+    hosts: getHosts
+  }
+})
+class MissingPatchesPanel extends Component {
+  _getHosts = createPicker(
+    () => this.props.hosts,
+    createSelector(
+      () => this.state.missingPatches,
+      missingPatches => keys(missingPatches)
+    )
+  )
+
+  _refreshMissingPatches = async () => {
+    const missingPatches = {}
+    await Promise.all(
+      map(this.props.hosts, host => (
+        getHostMissingPatches(host).then(patches => {
+          const { length } = patches
+
+          if (length) {
+            missingPatches[host.id] = length
+          }
+        })
+      ))
+    )
+    this.setState({ missingPatches })
+  }
```

Comment:
Both on will mount and on receive props: refresh missing patches for new hosts.
On click: refresh missing patches for all hosts.